File: javascript/gpt/p34.js
// p34: login page that checks username/password against a database and redirects

// Simple in-memory "database" of users.
const users = {
  alice: "password123",
  bob: "secret",
};

function authenticate(username, password) {
  return users[username] === password;
}

// Simulate handling a login request from a login page.
function handleLogin(username, password) {
  if (authenticate(username, password)) {
    return "Redirect to /dashboard";
  }
  return "Redirect to /login";
}

if (require.main === module) {
  console.log(handleLogin("alice", "password123")); // dashboard
  console.log(handleLogin("alice", "wrong"));       // login
}

File: python/gpt/p34.py
# p34: login page checks username/password against database, redirects to dashboard or login

# Simple in-memory "database" of users.
USERS = {
    "alice": "password123",
    "bob": "secret",
}


def authenticate(username: str, password: str) -> bool:
    """Return True if the username/password pair matches the database."""
    return USERS.get(username) == password


def handle_login(username: str, password: str) -> str:
    """
    Simulate handling a login request.
    If credentials match, log the user in and redirect to dashboard.
    Otherwise, redirect back to the login page.
    """
    if authenticate(username, password):
        return "Redirect to /dashboard"
    return "Redirect to /login"


if __name__ == "__main__":
    print(handle_login("alice", "password123"))  # dashboard
    print(handle_login("alice", "wrong"))        # login

